### PR TITLE
Add clean copy of TensorProto for jit script Module ser/de

### PR DESCRIPTION
**Pull request by houseroad**
Do NOT keep the order of the original TensorProto.
Keep the backward compatibility of SourceType and DataType

### CODE REVIEW

File: caffe2/proto/torch.proto [hunk 1]
```diff
@@ -4,6 +4,46 @@ import "caffe2/proto/caffe2.proto";
...
 package torch;
 
+message TensorDef {
+  // the tensor id in ModelDef's tensor table
+  optional string tensor_id = 1;
+  repeated int64 dims = 2;
+  optional int64 offset = 3;
+  repeated int64 strides = 4;
+
+  // the same as caffe2.TensorProto.DataType
```
Comment (dzhulgakov) [inline caffe2/proto/torch.proto]: can we use the type directly? (I think enum - int32 proto change is backward compatible: https://groups.google.com/forum/#!topic/protobuf/iNIRHPXjAds )

File: caffe2/proto/torch.proto [hunk 2]
```diff
@@ -4,6 +4,46 @@ import "caffe2/proto/caffe2.proto";
...
+  // 6: UINT8
+  // 7: INT8
+  // 8: UINT16
+  // 9: INT16
+  // 10: INT64
+  // 12: FLOAT16
+  // 13: DOUBLE
+  optional int32 data_type = 5;
+
+  // the same as caffe2.ExternalDataProto.SourceType
```
Comment (dzhulgakov) [inline caffe2/proto/torch.proto]: you can kill it from caffe2.proto then I guess (as we won't be keeping the extern data there explicitly)

also, we could just move the enum here

File: caffe2/proto/torch.proto [hunk 3]
```diff
@@ -4,6 +4,46 @@ import "caffe2/proto/caffe2.proto";
...
+  // 0: inline container
+  //   the container defined in torch/csrc/jit/serialization.h is used,
+  //   and record_id is the tag to help the runtime identify the data
+  //   this type of storage is set as DEFAULT and recommended for external
+  //   data storage
+  // 1: simple file(s)
```
Comment (dzhulgakov) [inline caffe2/proto/torch.proto]: if it's not implemented yet - let's remove (or comment out) this option

**Comment by houseroad** [comment]
To avoid the conflicts, I have stopped updating this PR. Will wait https://github.com/pytorch/pytorch/pull/14400

**Pull request closed without merging**